audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/rest/RestControllerContractTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.contract.audit.rest;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import uk.gov.gchq.palisade.service.audit.AuditApplication;
+import uk.gov.gchq.palisade.service.audit.request.AuditRequest;
+import uk.gov.gchq.palisade.service.audit.service.AuditService;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = AuditApplication.class, webEnvironment = WebEnvironment.DEFINED_PORT)
+public class RestControllerContractTest extends AuditTestCommon {
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+    @Autowired
+    private Map<String, AuditService> serviceMap;
+
+    List<AuditRequest> requests = List.of(
+            readRequestCompleteAuditRequest(),
+            readRequestExceptionAuditRequest(),
+            registerRequestCompleteAuditRequest(),
+            registerRequestExceptionAuditRequest()
+    );
+
+    @Test
+    public void contextLoads() {
+        assertNotNull(serviceMap);
+        assertNotEquals(serviceMap, Collections.emptyMap());
+    }
+
+    @Test
+    public void isUp() {
+        final ResponseEntity<String> health = restTemplate.getForEntity("/actuator/health", String.class);
+
+        assertThat(health.getStatusCode(), is(equalTo(HttpStatus.OK)));
```

Comment:
This test is not doing what it says on the tin.  The assertion tests for a HTTP 200 code.  The title of the method suggest that we are testing for the health indicator.  That would be a test to verify the body of the response includes the message {"status" : "UP"}.  The two are not necessary the same thing.

Reply:
all we need to test is the status as this is what k8s uses, not status:UP

Reply:
^^ k8s only cares that it returns 200 OK